addon/components/sl-modal.js

```diff
@@ -0,0 +1,236 @@
+
+    /**
+     * The modal service used to register and unregister this modal component
+     *
+     * @type {ember/Service}
+     */
+    modalService: Ember.inject.service( 'sl-modal' ),
+
+   /**
+    * The unique name use to register this modal with the modal service
+    *
+    * If this value is left null(default), then the component will not
+    * be registered on the modal service.
+    *
+    * @type {?String}
+    */
+    name: null,
+
+    /**
+     * tabindex attribute value
+     *
+     * @type {String}
+     */
+    tabindex: '-1',
+
+
+    // -----------------------------------------------------
+    // Observers
+
+
+    /**
+     * aria-labelledby attribute value
+     *
+     * @function
+     * @returns {String}
+     */
+    'aria-labelledby': Ember.computed(
+        'elementId',
+        function() {
+            return 'modalTitle' + this.get( 'elementId' );
+        }
+    ),
+
+    /**
+     * Register component on the modal service
+     *
+     * @function
+     * @returns {function}
+     */
+    register: Ember.on(
+       'init',
+       function() {
+           let name = this.get( 'name' );
+
+           if ( name ) {
+              this.get( 'modalService' ).register( this, name );
+           }
+       }
+    ),
+
+    /**
+     * Set up the component as a Bootstrap Modal and listen for events
+     *
+     * @function
+     * @returns {function}
+     */
+     setupModal: Ember.on(
+         'didInsertElement',
+         function() {
+            let modal = this.$().modal({
+                keyboard: true,
+                show: false,
+                backdrop: this.get( 'backdrop' )
+            });
+
+            modal.on( 'show.bs.modal', () => {
+               this.sendAction( 'beforeShow' );
+            });
+
+            modal.on( 'shown.bs.modal', () => {
+               this.set( 'isOpen', true );
+               this.sendAction( 'afterShow' );
+            });
+
+            modal.on( 'hide.bs.modal', () => {
+               this.sendAction( 'beforeHide' );
+            });
+
+            modal.on( 'hidden.bs.modal', () => {
+               this.set( 'isOpen', false );
+               this.sendAction( 'afterHide' );
+            });
+         }
+     ),
+
+    // -----------------------------------------------------
+    // Methods
+
+    show() {
+        this.$().modal( 'show' );
+    },
+
+    hide() {
+        this.$().modal( 'hide' );
+    },
+
+    /**
+     * Unbind bootstrap event handlers
+     * @function unbindHandlers
+     * @returns {undefined}
+     */
+    unbindHandlers: Ember.on(
+        'willDestroyElement',
+        function() {
+            this.$().off( 'show.bs.modal' );
+            this.$().off( 'shown.bs.modal' );
+            this.$().off( 'hide.bs.modal' );
+            this.$().off( 'hidden.bs.modal' );
+        }
+    ),
+
+    /**
+     * Unregister model from modelService
+     * @function unregister
+     * @returns {undefined}
+     */
+    unregister() {
```

Comment:
I cannot find any code calling this method.  Was is maybe meant to be called as a result of will or did DestroyElement?